streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -430,6 +449,10 @@ public void process(final ProcessorSupplier<K, V> processorSupplier, String... s
                                          Serde<V> thisValueSerde,
                                          Serde<V1> otherValueSerde,
                                          KStreamImplJoin join) {
+        Objects.requireNonNull(other, "other can't be null");
```

Comment:
The error messages would be a bit confusing for users who did not read javadocs, maybe renaming them to `the other join stream object can't be null` etc? Ditto below.